packages/main/src/Dialog.js

```diff
@@ -167,14 +193,22 @@ class Dialog extends Popup {
 		};
 	}
 
+	_clamp(val, min, max) {
+		return Math.min(Math.max(val, min), max);
+	}
+
 	onBeforeRendering() {
 		this.onPhone = isPhone();
 		this.onDesktop = isDesktop();
 	}
 
 	onEnterDOM() {
+		this._isRTL = this.effectiveDir === "rtl";
```

Comment:
Moved it to occur in onBeforeRendering. In the future, it might be better to do this check centrally (once) before all the ui5-webcomponents are rendered. I see a few web components are already doing this very same check, so it's fair to attempt to remove the duplication.